src/Hamiltonians/HOCartesian2BosonRelative.jl

```diff
@@ -0,0 +1,230 @@
+
+Implements a single particle in an arbitrary harmonic trap with a delta-function 
+at the centre, equivalent to the relative Hamiltonian of two bosons in a harmonic 
+oscillator with contact interaction (bare) interaction strength `g`,
+```math
+\\hat{H}_\\mathrm{rel} = \\sum_\\mathbf{i} ϵ_\\mathbf{i} + g\\sum_\\mathbf{ij} V_\\mathbf{ij} a^†_\\mathbf{i} a_\\mathbf{i}.
+```
+For a ``D``-dimensional harmonic oscillator indices ``\\mathbf{i}, \\mathbf{j}, \\ldots`` 
+are ``D``-tuples. The energy scale is defined by the first dimension i.e. ``\\hbar \\omega_x`` 
+so that single particle energies are 
+```math
+    \\frac{\\epsilon_\\mathbf{i}}{\\hbar \\omega_x} = (i_x + 1/2) + \\eta_y (i_y+1/2) + \\ldots.
+```
+The factors ``\\eta_y, \\ldots`` allow for anisotropic trapping geometries and are assumed to 
+be greater than `1` so that ``\\omega_x`` is the smallest trapping frequency.
+
+Matrix elements ``V_{\\mathbf{ij}}`` are for a contact interaction calculated in this basis
+```math
+    V_{\\mathbf{ij}} = \\prod_{d \\in x, y,\\ldots} \\psi_{i_d}(0) \\psi_{j_d}(0).
+```
+Only even parity states feel this interaction, so all ``i_d`` are even. Note that the matrix 
+representation of this Hamiltonian is completely dense in the even-parity subspace.
+
+# Arguments
+
+* `addr`: The starting address is required for the [`Hamiltonians`](@ref) interface,
+    but is overwritten for this Hamiltonians. A warning will be issued if the input address 
+    has the wrong number of particles (1) or modes (`prod(S)`). This warning can be disabled 
+    with keyword argument `warn_address = false`.
+* `S`: Tuple of the number of levels in each dimension, including the groundstate. The first 
+    dimension should be the largest and sets the energy scale of the system, ``\\hbar\\omega_x``.
+    The aspect ratios are determined from the other elements of `S`.
+    Defaults to a 1D spectrum with number of levels matching modes of `addr`.
+* `Sx`, `ηs`: Alternatively, provide the number of modes (including the groundstate) in the 
+    first dimension `Sx` and `D-1` aspect ratios for the other dimensions `ηs = (η_y, ...)`. 
+    The elements of `ηs` should be at least `1.0`. Providing `Sx` and `ηs` will redefine 
+    `S = (Sx, Sy, ...)`.
+* `g`: the (bare) isotropic interparticle interaction parameter. The value of `g` is assumed 
+    to be in trap units.
+* `interaction_only`: if set to `true` then the noninteracting single-particle terms are 
+    ignored. Useful if only energy shifts due to interactions are required.
+
+See also [`HOCartesianEnergyConserved`](@ref) and[`HOCartesianEnergyConservedPerDim`](@ref).
+"""
+struct HOCartesian2BosonRelative{
+    D,  # number of dimensions
+    A<:BoseFS
+} <: AbstractHamiltonian{Float64}
+    addr::A
+    S::NTuple{D,Int64}
+    aspect::NTuple{D,Float64}
+    vtable::Vector{Float64}  # interaction coefficients
+    g::Float64
+    interaction_only::Bool
```

Comment:
```suggestion
    impurity_only::Bool
```
In line with my other comments. Here, the delta function is an impurity, not an interaction.